code/game/machinery/kilm.dm

```diff
@@ -0,0 +1,70 @@
+/obj/machinery/kilm
+	name = "kilm"
+	desc = "A stone kilm, can be filled with logs for fuel."
+	icon = 'icons/obj/fireplace.dmi'
+	icon_state = "kilm"
+
+	use_power = NO_POWER_USE
+	density = TRUE
+
+	var/on = FALSE
+	var/filled = FALSE
+	var/charges = 0
+	var/making = null
+
+/obj/machinery/kilm/attackby(obj/item/T, mob/user)
+	if(istype(T, /obj/item/grown/log))
+		charges ++
+		qdel(T)
+
+/obj/machinery/kilm/attackby(obj/item/stack/ore/S, mob/user)
+	if(istype(S, /obj/item/stack/ore/glass))
+		if(S.amount <= 5)
+			user.show_message("<span class='notice'>You add the sand to the kilm.</span>", 1)
+			filled = TRUE
+			S.amount = (S.amount - 5)
+			if(S.amount < 5)
+				qdel(S)
+			else
+				user.show_message("<span class='notice'>You need a at lest five sand piles to make anything of use.</span>", 1)
+
+/obj/machinery/kilm/attack_hand(mob/living/carbon/user)
+	. = ..()
+	if(.)
+		return
+	if(charges == 0)
```

Comment:
I feel like this could be made a switch statement

Reply:
Good idea thank you

Reply:
The kilm is for the moment being scrapped do to incompetents on my part